Sources/Hypertext.swift

```diff
@@ -6,18 +6,23 @@
 //  Copyright © 2016 Sahand Nayebaziz. All rights reserved.
 //
 
-public protocol Renderable {
-    func render() -> String
-    func render(startingWithSpacesCount: Int) -> String
+public protocol Renderable: CustomStringConvertible {
+    func render(indent: Int?, startingWithSpacesCount: Int) -> String
```

Comment:
How does this work with two parameters? It looks like we only need one, right? And it's optional? I don't mind the idea of collapsing the render functions, I just want to make sure that at the point of use this is easy to use.

To me, hidden/optional parameters can be a bit hard to find and see, whereas I think seeing multiple render functions pop up in autocomplete when you start typing `ren..` is helpful and easy.